internal/xds/xdsclient/pool.go

```diff
@@ -99,7 +115,7 @@ func NewPool(config *bootstrap.Config) *Pool {
 // expected to invoke once they are done using the client.  It is safe for the
 // caller to invoke this close function multiple times.
 func (p *Pool) NewClient(name string, metricsRecorder estats.MetricsRecorder) (XDSClient, func(), error) {
-	return p.newRefCounted(name, metricsRecorder, defaultWatchExpiryTimeout)
+	return p.newRefCounted(name, metricsRecorder, defaultWatchExpiryTimeout, nil)
```

Comment:
If a pool was created with a non-nil config, i.e., by calling `NewPool(config)`, shouldn't this be passing that configuration instead of passing `nil`?

Reply:
Makes sense.